### ISSUE TITLE
GhostNet: More Features from Cheap Operations - 75.7% top-1 (better than than MobileNetV3)

### ISSUE THREAD

**Issue by AlexeyAB** [AlexeyAB/darknet#4418]
* paper: https://arxiv.org/abs/1911.11907v1
* source: https://github.com/iamhankai/ghostnet

maybe better than mobilenetv3, efficientnet, ..., etc.
![image](https://user-images.githubusercontent.com/12152972/69900881-f93fc080-13b3-11ea-8567-b6d9ca314fa1.png)

----

![image](https://user-images.githubusercontent.com/4096485/69901000-29736900-138c-11ea-800d-eab42763dd5c.png)

----

![image](https://user-images.githubusercontent.com/4096485/69901004-385a1b80-138c-11ea-9fd8-88bc7220e716.png)

**Comment by AlexeyAB** [comment]
@WongKinYiu Hi,
I added `ghostnet.cfg.txt` so you can try to train it for 600 000 iterations with batch_size=192 (mini_batch_size=96).

May be it can be fast on CPU/Neurochips (OpenCV-dnn).

**Comment by WongKinYiu** [comment]
@AlexeyAB thank you!

**Comment by AlexeyAB** [comment]
@WongKinYiu I just added dropout after avg-pooling. So if you already start training, you can download new cfg-file and continue training.

**Comment by duducode** [comment]
我在cpu度测试darnet-19 速度1.3秒一张图片，测试图片大小500*374，请问哪里出问题了

**Comment by WongKinYiu** [comment]
please use opencv dnn module instead of darknet for cpu testing.
https://github.com/AlexeyAB/darknet#yolo-v3-in-other-frameworks

**Comment by WongKinYiu** [comment]
Do you build darknet with (OPENMP=1 AVX=1)?

**Comment by duducode** [comment]
> Do you build darknet with (OPENMP=1 AVX=1)?
> And which CPU do you use?

thanks，i will try update my code and try again

**Comment by WongKinYiu** [comment]
@AlexeyAB 

top-1 1.5%, top-5 5.6%.

**Comment by AlexeyAB** [comment]
@WongKinYiu Thanks!
ghostnet.cfg.txt - 1.5% Top1? is it near ~0 ?
Can you share cfg/weights file?

**Comment by WongKinYiu** [comment]
@AlexeyAB 

[ghostnet.cfg.txt](https://github.com/AlexeyAB/darknet/files/4453572/ghostnet.cfg.txt)
[ghostnet_final.zip](https://github.com/AlexeyAB/darknet/files/4453574/ghostnet_final.zip)

**Comment by iamhankai** [comment]
@WongKinYiu This repo may help: https://github.com/d-li14/ghostnet.pytorch

**Comment by WongKinYiu** [comment]
@iamhankai thank you very much.

**Comment by rsek147** [comment]
@WongKinYiu I have tested with your cfg/weights. The result is almost same as yours.
I started training few days ago with almost same your cfg except for batch and subdivisions.
My result is top-1 30%, top-5 64% (300000 iterations, continuing). This is strange despite of using almost same cfg.

**Comment by AlexeyAB** [comment]
@WongKinYiu Thanks!

@rsek147 Can you attach your cfg-file?

**Comment by ouyanghuiyu** [comment]
@AlexeyAB @WongKinYiu I think the [ghostnet.cfg.txt](https://github.com/AlexeyAB/darknet/files/4453572/ghostnet.cfg.txt) is wrong，it can refer to https://github.com/d-li14/ghostnet.pytorch/blob/master/ghostnet.py , i use ghiost moudle in mobilenetv3 Small , it get 20% top1 after 20000 iters with 256 batch size

**Comment by maohu-96** [comment]
@AlexeyAB  hi！
When I use GhostNet for training, the loss -nan，How to solve？
I am completing a fruit test（class=1） using the ghostnet.cfg file above. What parts should I modify?
![image](https://user-images.githubusercontent.com/63272697/84853671-4f472d80-b092-11ea-90cd-ef3a38687243.png)

Thanks！

**Comment by wsypy** [comment]
@WongKinYiu Hi, have you been retraining Ghostnet since?If so, can you share .cfg and .weights files?

**Comment by WongKinYiu** [comment]
I did not get good result.
